Formula/crystal.rb

```diff
@@ -112,17 +127,16 @@ def install
     end
 
     # Install crystal
-    libexec.install ".build/crystal"
-    (bin/"crystal").write <<~SH
-      #!/bin/bash
-      EMBEDDED_CRYSTAL_PATH=$("#{libexec/"crystal"}" env CRYSTAL_PATH)
-      export CRYSTAL_PATH="${CRYSTAL_PATH:-"$EMBEDDED_CRYSTAL_PATH:#{prefix/"src"}"}"
-      export CRYSTAL_LIBRARY_PATH="${CRYSTAL_LIBRARY_PATH:+$CRYSTAL_LIBRARY_PATH:}#{HOMEBREW_PREFIX}/lib"
-      export PKG_CONFIG_PATH="${PKG_CONFIG_PATH:+$PKG_CONFIG_PATH:}#{Formula["openssl@1.1"].opt_lib/"pkgconfig"}"
-      exec "#{libexec/"crystal"}" "${@}"
-    SH
-
-    prefix.install "src"
+    crystal_install_dir.install ".build/crystal"
+    pkgshare.install "src"
+
+    embedded_crystal_path = "$(\"#{crystal_install_dir}/crystal\" env CRYSTAL_PATH)"
+    crystal_env = {
+      CRYSTAL_PATH:         "${CRYSTAL_PATH:-#{embedded_crystal_path}:#{pkgshare}/src}",
+      CRYSTAL_LIBRARY_PATH: "${CRYSTAL_LIBRARY_PATH:+${CRYSTAL_LIBRARY_PATH}:}#{HOMEBREW_PREFIX}/lib",
```

Comment:
These two path overrides should be unnecessary.
We're already configuring `CRYSTAL_CONFIG_LIBRARY_PATH` in the build step (see comment there to improve it).
In case we really need this dynamic override in the wrapper script, the `CRYSTAL_CONFIG_LIBRARY_PATH` configuration is useless and could be removed.

And we can bake in `CRYSTAL_PATH` exactly like `CRYSTAL_LIBRARY_PATH` with a relative path (see comment above).